lib/ExpensiMark.js

```diff
@@ -446,26 +446,22 @@ export default class ExpensiMark {
         let startIndex = 0;
 
         while (match !== null) {
-            // we want to avoid matching ending ) unless it is a closing parenthesis for the URL
-            if (textToCheck[(match.index + match[2].length) - 1] === ')' && !match[2].includes('(')) {
-                match[0] = match[0].substr(0, match[0].length - 1);
-                match[2] = match[2].substr(0, match[2].length - 1);
-            }
-
-            // remove extra ) parentheses
-            let brace = 0;
-            if (match[2][match[2].length - 1] === ')') {
-                for (let i = 0; i < match[2].length; i++) {
-                    if (match[2][i] === '(') {
-                        brace++;
-                    }
-                    if (match[2][i] === ')') {
-                        brace--;
-                    }
+            // We end the link at the last closing parenthesis that matches an opening parenthesis because unmatched closing parentheses are unlikely to be in the url
+            // and can be part of markdown for example
+            let unmatchedOpenParentheses = 0;
+            let url = match[2];
+            for (let i = 0; i < url.length; i++) {
+                if (url[i] === '(') {
+                    unmatchedOpenParentheses++;
                 }
-                if (brace) {
-                    match[0] = match[0].substr(0, match[0].length + brace);
-                    match[2] = match[2].substr(0, match[2].length + brace);
+                else if (url[i] === ')') {
+                    if(unmatchedOpenParentheses <= 0){ // unmatched closing parenthesis
```

Comment:
Comment should be above the code line, and the first letter capitalized
```suggestion
                    // We found an unmatched closing parenthesis
                    if(unmatchedOpenParentheses <= 0){ 
```